src/python/pants/backend/docker/goals/package_image_test.py

```diff
@@ -837,103 +838,30 @@ def test_invalid_build_target_stage(rule_runner: RuleRunner) -> None:
-    context_root: str | None,
-    options: dict | None,
-    expected_context_root: str,
+def test_get_context_root(
+    context_root: str | None, default_context_root: str, expected_context_root: str
 ) -> None:
-    rule_runner.write_files(
-        {
-            "src/docker/BUILD": f"docker_image(name='image', context_root={context_root!r})",
-            "src/docker/Dockerfile": "FROM python:3.8",
-        }
-    )
-
-    def check_docker_proc(process: Process):
-        assert process.argv == (
-            "/dummy/docker",
-            "build",
-            "--tag",
-            "image:latest",
-            "--file",
-            "src/docker/Dockerfile",
-            expected_context_root,
-        )
-
-    assert_build(
-        rule_runner,
-        Address("src/docker", target_name="image"),
-        options=options,
-        process_assertions=check_docker_proc,
-    )
-
-
-@pytest.mark.parametrize(
-    "context_root, options, raises",
-    [
-        (None, None, no_exception()),
-        (
-            "/abs/path",
-            None,
-            pytest.raises(
-                InvalidFieldException,
-                match=(
-                    r"The 'context_root' field in target src/docker:image must be a relative path, "
-                    r"but was '/abs/path'\."
-                ),
-            ),
-        ),
-        (
-            "BUILD:/abs/path",
-            None,
-            pytest.raises(
-                InvalidFieldException,
-                match=(
-                    r"The 'context_root' field in target src/docker:image must be a relative path, "
-                    r"but was '/abs/path'\."
-                ),
-            ),
-        ),
-        (
-            None,
-            {"default_context_root": "/abs/path"},
-            pytest.raises(
-                InvalidFieldException,
-                match=(
-                    r"The \[docker\]\.default_context_root configuration option must be a relative path, "
-                    r"but was '/abs/path'\."
-                ),
-            ),
-        ),
-    ],
-)
-def test_invalid_context_root(
-    rule_runner: RuleRunner, context_root: str | None, options: dict | None, raises: ContextManager
-) -> None:
-    rule_runner.write_files(
-        {
-            "src/docker/BUILD": f"docker_image(name='image', context_root={context_root!r})",
-            "src/docker/Dockerfile": "FROM python:3.8",
-        }
-    )
-
-    with raises:
-        assert_build(
-            rule_runner,
-            Address("src/docker", target_name="image"),
-            options=options,
-        )
+    docker_options = create_subsystem(
+        DockerOptions,
+        default_context_root=default_context_root,
+    )
+    address = Address("src/docker", target_name="image")
+    tgt = DockerImageTarget({"context_root": context_root}, address)
+    fs = DockerFieldSet.create(tgt)
+    assert fs.get_context_root(docker_options.default_context_root) == expected_context_root
```

Comment:
I noticed a great speed up by avoiding the use of the RuleRunner here..
This test takes ~2s to run, while the corresponding test using the RuleRunner took ~4s.